test/PoolFactory.t.sol

```diff
@@ -0,0 +1,177 @@
+        // Label addresses
+        vm.label(deployer, "Deployer");
+        vm.label(user1, "User 1");
+
+        vm.startPrank(deployer);
+
+        token1 = new Token1();
+        poolFactory = new PoolFactory();
+
+        vm.stopPrank();
+    }
+
+    function test_createPool() public {
+        // Set the initial data
+        address asset_ = address(token1);
+        address depositFeeRecipient_ = deployer;
+        uint32 baseRateTokensPerBlock_ = 20;
+        uint32 depositFee_ = 10;
+
+        // RewardsMultiplier
+        uint64[] memory rewardsMultiplierBlocks_;
+        uint64[] memory rewardsMultipliers_;
+
+        // WithdrawFees
+        uint64[] memory withdrawFeeBlocks_;
+        uint64[] memory withdrawFees_;
+
+        // Unhappy path Nº1 - Trying to create a Pool without being the Owner.
+        vm.startPrank(user1);
+
+        vm.expectRevert("Ownable: caller is not the owner");
+        poolFactory.createPool(
+            asset_,
+            depositFeeRecipient_,
+            baseRateTokensPerBlock_,
+            depositFee_,
+            rewardsMultiplierBlocks_,
+            rewardsMultipliers_,
+            withdrawFeeBlocks_,
+            withdrawFees_
+        );
+
+        vm.stopPrank();
+
+        vm.startPrank(deployer);
+        // Unhappy path Nº2 - Trying to create a Pool without passing reward multipliers.
+        vm.expectRevert(abi.encodeWithSignature("Pool_RewardMultipliersAmountZeroError()"));
+        poolFactory.createPool(
+            asset_,
+            depositFeeRecipient_,
+            baseRateTokensPerBlock_,
+            depositFee_,
+            rewardsMultiplierBlocks_,
+            rewardsMultipliers_,
+            withdrawFeeBlocks_,
+            withdrawFees_
+        );
+        rewardsMultiplierBlocks_ = new uint64[](4);
+        rewardsMultiplierBlocks_[0] = 100;
+        rewardsMultiplierBlocks_[1] = 200;
+        rewardsMultiplierBlocks_[2] = 300;
+        rewardsMultiplierBlocks_[3] = 400;
+        rewardsMultipliers_ = new uint64[](3);
+
+        // Unhappy path Nº3 - Trying to create a Pool without passing the same number of reward multipliers parameters.
+        vm.expectRevert(abi.encodeWithSignature("Pool_InitializeRewardMultipliersParametersFormatError()"));
+        poolFactory.createPool(
+            asset_,
+            depositFeeRecipient_,
+            baseRateTokensPerBlock_,
+            depositFee_,
+            rewardsMultiplierBlocks_,
+            rewardsMultipliers_,
+            withdrawFeeBlocks_,
+            withdrawFees_
+        );
+        rewardsMultipliers_ = new uint64[](4);
+        rewardsMultipliers_[0] = 100;
+        rewardsMultipliers_[1] = 50;
+        rewardsMultipliers_[2] = 25;
+        rewardsMultipliers_[3] = 10;
+
+        // Unhappy path Nº4 - Trying to create a Pool without passing withdraw fees.
+        vm.expectRevert(abi.encodeWithSignature("Pool_WithdrawFeesAmountZeroError()"));
+        poolFactory.createPool(
```

Comment:
info: create auxiliary createPool internal function, it's okay since many arguments are storage variables and you can probably fit it in one line